apps/app.go

```diff
@@ -162,14 +167,22 @@ type indexedPortDefinition struct {
 }
 
 func (app App) findConsulPortDefinitions() []indexedPortDefinition {
+	var appPortDefinitions []PortDefinition
+	if len(app.Container.PortMappings) > 0 {
```

Comment:
Can you add a comment here with information about deprecation of `PortDefinitons` in favour of `PortMappings` and link to Marathon release notes?

Reply:
We can extract this part to a function with proper godoc. e.g.

```go
func (app App) getPortDefinitions() []PortDefinition {
	var appPortDefinitions []PortDefinition
        if len(app.Container.PortMappings) > 0 {
		appPortDefinitions = app.Container.PortMappings
 	} else {
 		appPortDefinitions = app.PortDefinitions
 	}
        return appPortDefinitions
}
```

Reply:
Can you add a test for this change?

Reply:
Sounds good. Added the changes and unittest